CovertInfiltration/Src/CovertInfiltration/Classes/CI_UIStrategyMapItem_CovertAction.uc

```diff
@@ -1,6 +1,102 @@
+//---------------------------------------------------------------------------------------
+//  AUTHOR:  ArcaneData
+//  PURPOSE: MCO for covert action items on strategy map, adding the infiltration nameplate
+//---------------------------------------------------------------------------------------
+//  WOTCStrategyOverhaul Team
+//---------------------------------------------------------------------------------------
 class CI_UIStrategyMapItem_CovertAction extends UIStrategyMapItem_CovertAction;
 
+var UIScanButton ScanButton;
+var UIText PercentLabel;
+var UIText ProgressLabel;
+
+var transient bool bScanButtonResized;
+var transient float CachedScanButtonWidth;
+
+var localized string strProgress;
+var localized string strCovertAction;
+
+simulated function UIStrategyMapItem InitMapItem(out XComGameState_GeoscapeEntity Entity)
+{
+	local float ScanWidth;
+
+	super.InitMapItem(Entity);
+
+	ScanButton = Spawn(class'UIScanButton', self).InitScanButton();
+	ScanButton.SetButtonIcon("");
+	//ScanButton.SetDefaultDelegate(OpenInfiltrationMissionScreen);
+	ScanButton.SetButtonType(eUIScanButtonType_Default);
+
+	PercentLabel = Spawn(class'UIText', ScanButton).InitText('PercentLabel', "");
+	PercentLabel.SetWidth(60); 
+	PercentLabel.SetPosition(154, 3);
+
+	ProgressLabel = Spawn(class'UIText', ScanButton).InitText('ProgressLabel', class'UIUtilities_Text'.static.GetSizedText(strProgress, 12));
+	ProgressLabel.SetWidth(60); 
+	ProgressLabel.SetPosition(154 + 15, 23);
+	
+	bScanButtonResized = false;
+
+	return self;
+}
+
+function UpdateFromGeoscapeEntity(const out XComGameState_GeoscapeEntity GeoscapeEntity)
+{
+	local int InfilPercent;
+	local float ScanWidth;
+	local XComGameState_CovertAction CovertAction;
+	local float TotalDuration, RemainingDuration;
+
+	if( !bIsInited ) return;
+
+	super(UIStrategyMapItem).UpdateFromGeoscapeEntity(GeoscapeEntity);
```

Comment:
No real need for `(UIStrategyMapItem)` - `UIStrategyMapItem_CovertAction` is basically empty